src/features.jl

```diff
@@ -0,0 +1,74 @@
+equalvolumeradius(radii) = sum(radii.^3)^(1/3)
+
+function rescast(f, res::PDBResidue)
+    return [f(a,res.id.name) for a in res.atoms]
+end
+
+function add_feature!(mgmm::IsotropicMultiGMM, key, μ, σ, ϕ)
+    if !haskey(mgmm.gmms, key)
+        push!(mgmm.gmms, key => IsotropicGMM([IsotropicGaussian(μ, σ, ϕ)]))
+    else
+        push!(mgmm.gmms[key].gaussians, IsotropicGaussian(μ, σ, ϕ))
+    end
+end
+
+function add_features_from_atom!(mgmm::IsotropicMultiGMM, a::PDBAtom, r::PDBResidue, ϕ=1.0)
+    σ = vanderwaalsradius[(r.id.name, a.atom)]
+    if ishydrophobic(a, r.id.name)
+        add_feature!(mgmm, :Hydrophobe, a.coordinates, σ, ϕ)
+    end
+    if isaromatic(a, r.id.name)
+        add_feature!(mgmm, :Aromatic, a.coordinates, σ, ϕ)
+    end
+    if iscationic(a, r.id.name)
+        add_feature!(mgmm, :PosIonizable, a.coordinates, σ, ϕ)
+    end
+    if isanionic(a, r.id.name)
+        add_feature!(mgmm, :NegIonizable, a.coordinates, σ, ϕ)
+    end
+    if ishbonddonor(a, r.id.name)
+        add_feature!(mgmm, :Donor, a.coordinates, σ, ϕ)
+    end
+    if ishbondacceptor(a, r.id.name)
+        add_feature!(mgmm, :Acceptor, a.coordinates, σ, ϕ)
+    end
+end
+
+function add_features_from_atoms!(mgmm::IsotropicMultiGMM, key, atoms, r, ϕ=1.0)
+    if isempty(atoms)
+        return
+    end
+    μ = mean(a -> a.coordinates, atoms)
+    σ = equalvolumeradius([vanderwaalsradius[(r.id.name, a.atom)] for a in atoms])
+    add_feature!(mgmm, key, μ, σ, ϕ)
```

Comment:
IIUC, this adds a single atom-sized feature at the center of mass to stand for all the features. What about doing something like setting `σ = std(vec(a.coordinates .- μ)) + equalvolumeradius([vanderwaalsradius[(r.id.name, a.atom)] for a in atoms])`? That would take into account the distribution of the list of atoms (whether that's a good or bad thing, I'm less sure). And if `ϕ` is the amplitude, would it be better to use a default of `ϕ = length(atoms)`?

Is there any meaningful comparison with [`aa_properties`](https://github.com/HolyLab/GPCRAnalysis.jl/blob/fd7d322fb7b8257920173d58f46a04122b40df85/src/properties.jl#L10-L31)? Seems like they should be correlated somehow? Do we have greater confidence in doing it this way, or would it be better to somehow convert an `AAProperties` into a GMM?

Reply:
I think there is an information tradeoff in switching to using `aa_proprties`, since we lose some information about position and locations of donor/acceptor/aromatic groups. 

If we wanted to build GMM features using these, would we just make "charge" and "hydropathy" features at the sidechain centroids, with `σ` coming from the volume and and `ϕ` coming from `charge` or `hydropathy` respectively? This should be pretty straightforward to do already if we want to give this a try in a script, since we don't have to bother with iterating over atoms.